backend-worker/ucm_valuation.py

```diff
@@ -456,41 +452,74 @@ def grouped_scalar_calculation(
 
     # Calculate kWh map
     grouped_scalar_op = MultiplyRasterByScalarList(
         category_list, scalar_list, base_raster_nodata, category_raster_nodata
     )
 
-    temp_base_path = temporary_working_dir / Path(base_raster_path).name
-    temp_category_path = temporary_working_dir / Path(category_raster_path).name
-
-    pygeoprocessing.align_and_resize_raster_stack(
-        [str(base_raster_path), str(category_raster_path)],
-        [str(temp_base_path), str(temp_category_path)],
-        ["near", "near"],
-        (cell_size, -cell_size),
-        "intersection",
-    )
-
     pygeoprocessing.raster_calculator(
-        [(str(temp_base_path), 1), (str(temp_category_path), 1)],
+        [(base_raster_path, 1), (category_raster_path, 1)],
         grouped_scalar_op,
-        str(target_raster_path),
+        target_raster_path,
         gdal.GDT_Float32,
         base_raster_nodata,
     )
+
+
+def _add_zonal_stats_dict_to_vector(
+        vector_path, stats_map, field_name, aggregate_field_id):
+    """Add a new field to a shapefile with values from a dictionary.
+
+    Args:
+        vector_path (string): a path to a vector whose FIDs
+            correspond with the keys in `stats_map`.
+        stats_map (dict): a dictionary in the format generated by
+            pygeoprocessing.zonal_statistics that contains at least the key
+            value of `aggregate_field_id` per feature id.
+        field_name (str): a string for the name of the new field to add to
+            the target vector.
+        aggregate_field_id (string): one of 'min' 'max' 'sum' 'mean' 'count'
+            or 'nodata_count' as defined by pygeoprocessing.zonal_statistics.
+
+    Returns:
+        None
+
+    """
+    vector = gdal.OpenEx(
+        vector_path, gdal.OF_VECTOR | gdal.GA_Update)
+    layer = vector.GetLayer()
+    for idx, field in enumerate(layer.schema):
+        if field.name == field_name:
+            layer.DeleteField(idx)
+
+    # Create the new field
+    field_defn = ogr.FieldDefn(field_name, ogr.OFTReal)
+    field_defn.SetWidth(24)
+    field_defn.SetPrecision(11)
+    layer.CreateField(field_defn)
+
+    # Get the number of features (polygons) and iterate through each
+    layer.ResetReading()
+    for feature in layer:
+        feature_fid = feature.GetFID()
+
+        # Using the unique feature ID, index into the
+        # dictionary to get the corresponding value
+        # only write a value if zonal stats found valid pixels in the polygon:
+        if stats_map[feature_fid]['count'] > 0:
+            if aggregate_field_id == 'mean':
+                field_val = float(
+                    stats_map[feature_fid]['sum']) / stats_map[feature_fid]['count']
+            else:
+                field_val = float(stats_map[feature_fid][aggregate_field_id])
+
+            # Set the value for the new field
+            feature.SetField(field_name, field_val)
+
+            layer.SetFeature(feature)
```

Comment:
I'm not sure if this is necessary but we've typically set `layer` and `vector` to `None` before exiting the function as a precaution to having any hanging files open. I'm not sure if this is still an issue with later versions of GDAL.

Reply:
I don't know if it is necessary either. I took this function wholesale from invest, where we're not cleaning them up. But I have pushed a change to clean them up here.